### ISSUE TITLE
proposal to change title of this spec to include "integration"

### ISSUE THREAD

**Issue by jpiesing** [w3c/webmediaporting#1]
IMHO this spec is more about the integration of an HTML5 UA onto a media device and not just porting. I believe some people will ignore a document called only "porting" thinking it's not relevant to them when actually it is. I don't mind if we call it "porting and integration" if people think having "porting" in the title is important but I believe it's important to have "integration" in the title as well.

**Comment by mavgit** [comment]
I'm happy to use whatever is the term of art, whether "Porting", "Integration", "Porting and Integration" or some other term.

It would be nice to find what terminology is used by the Chromium, Edge, Gecko and WebKit projects. If there's common terminology in those projects, let's just follow that. Can anyone from these project teams speak to that?

I did some quick, limited searching and found usage of both terms (emphasis added):

> Portability
The WebKit project seeks to address a variety of needs. We want to make it reasonable to **_port_** WebKit to a variety of desktop, mobile, embedded and other platforms. We will provide the infrastructure to do this with tight platform **_integration_**, reusing native platform services where appropriate and providing friendly embedding APIs.
>
>Usability
To the extent that WebKit features affect the user experience, we want them to work in accordance with good human interface design principles, and to mesh well with platform-native HI conventions. Furthermore, we strive to **_integrate_** with platform accessibility features to allow access for all users, including those with disabilities.
https://webkit.org/project/

>Gecko runs today on Win32 (Windows XP Service Pack 2, Windows Vista, Windows 7, Windows 8, Windows 8.1), Mac OS X 10.5 and later, and Linux. OEMs and contributors from the Net participating in mozilla.org are **_porting_** Gecko to other platforms. Such **_porting_** efforts are underway for Solaris, HP/UX, AIX, Irix, OS/2, OpenVMS, BeOS, and Amiga, among others.
https://developer.mozilla.org/en-US/docs/Gecko/FAQ

> Chromium is a large and complex cross-platform product. We try to share as much code as possible between platforms, while implementing the UI and OS **_integration_** in the most appropriate way for each. While this gives a better user experience, it adds extra complexity to the code. This document describes the recommended practices for keeping such cross-platform code clean.
> ...
> When **_porting_**, consider converting cases of the latter model to the former model if the UI element is something simple like a dialog box.
https://www.chromium.org/developers/design-documents/conventions-and-patterns-for-multi-platform-development

> Development Guides
> ...
> Linux Development tips and **_porting_** guide
https://www.chromium.org/developers

> Ozone is a meta-platform (http://www.chromium.org/developers/design-documents/ozone) for **_porting_** Chromium to new platforms that would otherwise require landing a new platform in the chromium tree.
This list is for ozone-related discussions, change notifications, **_porting_** help, etc.
https://groups.google.com/a/chromium.org/forum/#!forum/ozone-dev

**Comment by michelnev** [comment]
I agree with adding Integration to the title.  The way I've always viewed it is:

- Porting is the process of getting a software package (application) compiling/running on the device.
- Integration means pulling in the ported software package into the device's software stack.

**Comment by jpiesing** [comment]
> I agree with adding Integration to the title. The way I've always viewed it is:
> 
>     Porting is the process of getting a software package (application) compiling/running on the device.
>     Integration means pulling in the ported software package into the device's software stack.

+1

**Comment by mavgit** [comment]
I also agree with the definitions of porting and integration above. Given those definitions, it seems clear to me that all of our spec will be related to integration and none will be about actual porting (i.e. "compiling/running on the device").

So, is the consensus for "Porting and Integration" or for simply for switching to "Integration"?

**Comment by michelnev** [comment]
Following up from today's call, I'd vote for simply using "Integration"

**Comment by michelnev** [comment]
Can we do a call for consensus on this one?

**Comment by mavgit** [comment]
I agree we have consensus that "Integration" is better than "Porting and Integration".

So, eliminating any mention of "porting", should the full spec title be:
>Web Media User Agent Integration

or something else?
